// ai_service.proto
syntax = "proto3";

package ai;

service AIService {
  rpc GetSessionIds (SessionIDsRequest) returns (SessionIDsResponse) {}
  rpc GetChat (GetChatRequest) returns (GetChatResponse) {}
  rpc GetHistory (GetHistoryRequest) returns (GetHistoryResponse) {}
  rpc FirstChat (FirstChatRequest) returns (FirstChatResponse) {}
}

message ChatMessage {
  string role = 1;
  string text = 2;
}


message SessionIDsRequest {
  string user_id = 1;
}

message SessionIDsResponse {
  string session_ids = 1;
}

message GetHistoryRequest {
  string user_id = 1; 
  string session_id = 2;
}

message GetHistoryResponse {
  ChatMessage history = 1;
}

message GetChatRequest {
  string user_id = 1; 
  string session_id = 2;
  string question = 3;
}

message GetChatResponse {
  
  string response = 1;
}

message FirstChatRequest {
  string user_id = 1; 
  string question = 2;
}

message FirstChatResponse {
  string response = 1;
}